test/integration_test.rb

```diff
@@ -0,0 +1,62 @@
+require 'helpers/kubeclient_helper'
+
+module KubernetesDeploy
+  class IntegrationTest < ::Minitest::Test
+    include KubeclientHelper
+
+    def setup
+      @logger_stream = StringIO.new
+      @logger = Logger.new(@logger_stream)
+      KubernetesDeploy.logger = @logger
+      @namespace = TestProvisioner.claim_namespace
+    end
+
+    def teardown
+      @logger_stream.close
+      TestProvisioner.delete_namespace(@namespace)
+    end
+  end
+
+  module TestProvisioner
+    extend KubeclientHelper
+
+    def self.claim_namespace
```

Comment:
The idea here is to let each test get its own unique namespace so they don't leak k8s state. We'll for sure want to test redeploys, but IMO that should be within one test. Suggestions for better name generation welcome (can we grab part of the test name somehow?). Why not use one namespace and wait for it to be deleted in teardown? Because namespace deletion sometimes takes many seconds.